src/cpp/core/include/core/http/AsyncConnectionImpl.hpp

```diff
@@ -95,6 +95,11 @@ class AsyncConnectionImpl :
       if (close)
          response_.setHeader("Connection", "close");
 
+      // make sure that if no content-length was specified, we send a 0-length one
+      // otherwise, this response will be invalid
+      if (response_.headerValue("Content-Length").empty())
```

Comment:
Let's ensure that the body is empty before we do this -- on casual inspection it looks like we typically set the content-length when adding a body, but I couldn't prove that this is always the case.